graphql/resolvers/Author.js

```diff
@@ -0,0 +1,93 @@
+import { Op } from 'sequelize';
+import { Author } from '../../db-schema/postgres/Author.js';
+import { AuthorDetail } from '../../db-schema/mongo/AuthorDetail.js';
+
+const batchAuthors = async (authorIds) => {
+    const authors = await Author.findAll({
+      where: {
+        id: {
+            [Op.in]: authorIds,
+        },
+      },
+    });
+    return authorIds.map(id => authors.find(author => author.id === id) || null);
+};
+
+const AuthorResolver = {
+    Author: {
+        books: async (author, _, context) => {
+            return context.loaders.book.load(author.id);
+        },
+    },
+    Query: {
+        authors: async (_, { filter, page = 1, pageSize = 10 }) => {
+            const whereClause = {};
+            
+            if (filter) {
+                if (filter.name) {
+                  whereClause.name = { [Op.iLike]: `%${filter.name}%` };
+                }
+                if (filter.id) {
+                  whereClause.id = filter.id;
+                }
+                if (filter.born_date_gt) {
+                  whereClause.born_date = { ...whereClause.born_date, [Op.gte]: new Date(filter.born_date_gt) };
+                }
+                if (filter.born_date_lt) {
+                  whereClause.born_date = { ...whereClause.born_date, [Op.lte]: new Date(filter.born_date_lt) };
+                }
+                if (filter.born_date_range) {
+                  whereClause.born_date = { 
+                    [Op.between]: [
+                      new Date(filter.born_date_range.start), 
+                      new Date(filter.born_date_range.end)
+                    ] 
+                  };
+                }
+            }
+
+            const offset = (page - 1) * pageSize;
+            
+            const { rows, count } = await Author.findAndCountAll({
+                where: whereClause,
+                limit: pageSize,
+                offset: offset,
+            });
+
+            return {
+                edges: rows,
+                pageInfo: {
+                    hasNextPage: offset + rows.length < count,
```

Comment:
`hasNextPage: offset + rows.length < count` this logic will get duplicated, anyway to create a common function which can serve purpose for paginated query?

Reply:
Initially I started with the logic that because books will be more and authors will be less, so I will try to have a diff pagination logic for both of them, but ended up keeping it the same because of time constraints. But yes, I understand how reusing would make sense.